MediaBrowser.Providers/Plugins/Tmdb/Movies/TmdbSearch.cs

```diff
@@ -163,17 +164,36 @@ private async Task<List<RemoteSearchResult>> GetSearchResultsGeneric(string name
         {
             if (string.IsNullOrWhiteSpace(name))
             {
-                throw new ArgumentException("name");
+                throw new ArgumentException("String can't be null or empty.", nameof(name));
             }
 
-            var url3 = string.Format(_searchURL, WebUtility.UrlEncode(name), TmdbUtils.ApiKey, language, type);
+            string url3;
+            if (year != null && string.Equals(type, "movie", StringComparison.OrdinalIgnoreCase))
+            {
+                url3 = string.Format(
+                    CultureInfo.InvariantCulture,
+                    SearchUrl,
+                    WebUtility.UrlEncode(name),
+                    TmdbUtils.ApiKey,
+                    language,
+                    type) + "&primary_release_year=" + year;
```

Comment:
```suggestion
                url3 = string.Format(
                    CultureInfo.InvariantCulture,
                    SearchUrlWithYear,
                    WebUtility.UrlEncode(name),
                    TmdbUtils.ApiKey,
                    language,
                    type,
                    year);
```

Reply:
`first_air_date_year` != `primary_release_year`

Reply:
Read it too quickly! Then there should be another SearchUrl?